### PR TITLE
Persistent fingerprints + GMS2.3 support

### PR DESCRIPTION
**Pull request by EnbyAce**
The original function generated a fingerprint based that may change frequently, for example the count of connected gamepads. My fork doesn't do that, but has a more persistent value, the player's OS's language.

Besides that, my fork is made to support the way functions are written in GameMaker Studio 2.3 and onward and it uses a Linux-only environment variable, the username.

### CODE REVIEW

**Comment by silbinarywolf** [comment]
Heya! 

This looks pretty good at a glance but I'm not really interested in maintaining this anymore.
ie. Spending my free time pulling this code down, testing it on the various operating systems and okaying it.

I think instead of merging your pull request in, I'd rather update my README to point to your repository as the newer thing and simply archive this repository so it becomes read-only. But that's only if you feel comfortable with me doing that!

In terms of high-level code reviewing of this, some minor concerns I have are as follows:

- Removal of code comments telling users of the code what values certain environment variables return. I think documenting these in comments is worth it so folks can see at a glance what that data looks like. (Without needing to check themselves)
  - ie. `"// Add computer name (example value: DESKTOP-GH1TA)"`
- Removal of `os_browser == browser_not_a_browser` in the if-statement checks. This was added so fingerprint would work in the browser as well and ignore trying to get environment variables within a browser/HTML5 environment.

In anycase, it's your code now as far as I'm concerned so, these are only suggestions :)

**tldr:** If you're fine with me updating my README.md to point to your repository, closing this PR and making my repository read-only, then I'll do that.

**Comment by EnbyAce** [comment]
I'm fine with the README pointing to my fork. However, there's still the [marketplace link](https://marketplace.yoyogames.com/assets/6613/fingerprint) and I personally won't be able to maintain that.

**Comment by silbinarywolf** [comment]
Ah thanks for the reminder! I completely forgot that the GMS Marketplace existed.

I'm not sure what to do there, I've at least updated the description to point to your Github repository as well but my only other options there are to make the asset "private" (only accessible if you have a link to it) or to take it "off-sale" (only downloadable if you got it previously), neither of which feel like a good idea.

If what I've done feels good enough to you, then I'll leave it there and close this PR.

**Comment by EnbyAce** [comment]
The new README looks fine to me, is it possible to transfer ownership of the asset on the YoYoGames Marketplace to me though?

**Comment by silbinarywolf** [comment]
There's seemingly no way to do it in the user interface. However I can try raising a support ticket about it. Got a publisher account I can link to in the support ticket?

For example, mine is: https://marketplace.yoyogames.com/publishers/286/silbinarywolf

**Comment by EnbyAce** [comment]
I made [one](https://marketplace.yoyogames.com/publishers/5960/acekiron) earlier today, but I can't access it anymore. So I made [a new one](https://marketplace.yoyogames.com/publishers/5961/acekiron).

**Comment by silbinarywolf** [comment]
Awesome, I've created a support ticket saying the following. We'll see what they say back.

> Hi there,
> 
> I was hoping to transfer ownership of my Fingerprint asset to another user/publisher. Is this possible? If so, are you able to move my Fingerprint Asset linked below to the following publisher?
> 
> Fingerprint Asset:
> https://marketplace.yoyogames.com/assets/6613/fingerprint
> 
> Publisher to transfer to:
> https://marketplace.yoyogames.com/publishers/5961/acekiron-new-profile
> 
> Both of us discussing transference of ownership here:
> https://github.com/silbinarywolf/fingerprint-gml/pull/1
> 
> Kind regards, Jae

**Comment by EnbyAce** [comment]
Ok, thanks

**Comment by silbinarywolf** [comment]
I got the following reply and I'm not gonna pursue it further. If you put up a Marketplace link, I'll look at just taking mine down.

> Hi,
> 
> Thank you for contacting us.
> 
> There is no automated way to transfer the ownership of an asset.
> 
> I would advise uploading the asset on the new publisher profile.
> 
> Let me know if you have any other questions or issues.
>

**Comment by AceKiron** [comment]
It's requesting yymanifest.xml and metadata.json files.

**Comment by EnbyAce** [comment]
I've just put it on the Marketplace. https://marketplace.yoyogames.com/assets/10114/fingerprint-revamped

**Comment by silbinarywolf** [comment]
I've put my asset "Off sale". Folks that downloaded it already can get it but it should stop new downloads and folks can find yours.

If this is all good to you, I'll close this PR now.

image

**Comment by EnbyAce** [comment]
It seems ok to me
